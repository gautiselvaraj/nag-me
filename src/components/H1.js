import React from 'react';
import styled from 'styled-components';

const Heading = styled.h1`
  font-size: 1.25rem;
  margin-top: 0;
  margin-bottom: 0;
`;

export default ({children}) => <Heading>{children}</Heading>
