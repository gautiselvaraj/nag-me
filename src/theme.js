export default {
  main:           '#214154',
  mainDarken:     '#083148',
  mainLighten:    '#215a7b',
  black:          '#111111',
  white:          '#f6f6f6',
  greyLightest:   '#efefef',
  greyLighter:    '#dddddd',
  greyLight:      '#a5a5a5',
  grey:           '#898989',
  greyDark:       '#777777',
  greyDarker:     '#595959',
  greyDarkest:    '#222',
}
