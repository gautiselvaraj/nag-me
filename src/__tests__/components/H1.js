import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import H1 from '../../components/H1';
import theme from '../../theme';
import 'jest-styled-components';

describe('<H1 />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<H1 theme={theme}>Heading 1</H1>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have correct text, element & color', () => {
    const h1 = mount(<H1 theme={theme}>Heading 1</H1>);
    expect(h1.text()).toBe('Heading 1');
    expect(h1.find('h1').length).toBe(1);
    expect(h1).toHaveStyleRule('color', theme.white);
  });
});
