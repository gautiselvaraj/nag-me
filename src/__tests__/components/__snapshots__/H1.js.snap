// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H1 /> should match snapshot 1`] = `
.c0 {
  color: #f6f6f6;
  font-size: 1.25rem;
  margin-top: 0;
  margin-bottom: 0;
}

<h1
  className="c0"
>
  Heading 1
</h1>
`;
